src/cfnlint/__init__.py

```diff
@@ -1327,28 +1327,22 @@ def __init__(
     def transform(self):
         """Transform logic"""
         LOGGER.debug('Transform templates if needed')
-
+        sam_transform = 'AWS::Serverless-2016-10-31'
         matches = []
         transform_type = self.cfn.template.get('Transform')
-
         # Don't call transformation if Transform is not specified to prevent
         # useless execution of the transformation.
         # Currently locked in to SAM specific
         if transform_type:
-            if isinstance(transform_type, list):
-                if len(transform_type) != 1:
-                    return matches
-                if transform_type[0] != 'AWS::Serverless-2016-10-31':
-                    return matches
-            if transform_type != 'AWS::Serverless-2016-10-31':
+            if isinstance(transform_type, list) and sam_transform not in transform_type:
+                return matches
+            if transform_type != sam_transform:
```

Comment:
I think you want this
```
if isinstance(transform_type, six.string_types) and transform_type != sam_transform:
                return matches
``` 

other wise the list won't match the string.

Reply:
To give more to this when you have this.
```
Transform: 
  - UnsupportedTransform
  - AWS::Serverless-2016-10-31
```

The first `if isinstance(transform_type, list):` won't return matches because it found `AWS::Serverless-2016-10-31` which means it hits the second if.  Since the list above won't match `AWS::Serverless-2016-10-31` ti returns matches and doesn't do the transform.  I believe your second `if` is to cover the case when its just a string so we should include that check as well.  Thoughts?

Reply:
🤦‍♂  Yeah you are right 🤦‍♂

Reply:
@kddejong 
Are you good with the change to normalize the transform_type check as a list to avoid the type checking below? It feels cleaner to me, but i can just add the `isinstance(transform_type, six.string_types)` if that is more inline with project conventions

Reply:
Yea I'm good with that.